sftp/sftp.go

```diff
@@ -0,0 +1,101 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// FileManager interface abstracts the SFTP client
```

Comment:
I find this comment a bit misleading. This interface doesn't abstract the SFTP client.
The SFTP client has a different set of methods which are:

```go
	Create(path string) (io.WriteCloser, error)
	Open(path string) (io.ReadCloser, error)
	Remove(path string) error
```

```suggestion
// FileManager is an interface for managing files on a remote server
```

Reply:
addressed